File: RoomExercise/app/src/main/java/com/example/roomexercise/EmployeeDao.java
package com.example.roomexercise;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface EmployeeDao {

    @Query("SELECT * from Employee")
    List<EmployeeEntity> getEmployeeList();

    @Insert
    void insert(EmployeeEntity employeeEntity);

    @Delete
    void delete(EmployeeEntity employeeEntity);

    @Update
    void update(EmployeeEntity employeeEntity);


}

File: RoomExample/app/src/main/java/com/example/roomexample/Room/EmployeeDao.kt
package com.example.roomexample.Room

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.example.roomexample.Room.Employee

@Dao

interface EmployeeDao{
    @Query("SELECT * FROM Employee")
    fun listAll():LiveData<List<Employee>>
    @Insert
    fun insertAll(vararg employee: Employee)
    @Delete
    fun delete(employee: Employee)

    @Query("SELECT * FROM EMPLOYEE WHERE firstName like :firstName")
    fun getByName(firstName:String): LiveData<Employee>
}